### ISSUE TITLE
Class similarity weighted KD makes no differences

### ISSUE THREAD

**Issue by Ze-Yang** [HieuPhan33/REMINDER#4]
Hi, I tried to run without your proposed csw loss by removing ` --csw_kd ${LOSS} --delta_csw 1.0`. Strangely, I obtained almost the same results on ADE20k across different settings. Could you advice?

**Comment by HieuPhan33** [comment]
Hi

I find that CSW-KD is good for 100-5 and 100-10. Can you share the results on these two settings?

My GPU is occupied with other tasks. I will re-run experiments and come back to you later.

**Comment by Ze-Yang** [comment]
Here are the results on 100-10. For 100-5, I have not yet run.
|| 0-100 | 101-150 | all |
|--|:--:|:--:|:--:|
| w/ csw | 38.71 | 20.33 | 32.63 |
| w/o csw | 38.64 | 20.58 | 32.66 |

**Comment by HieuPhan33** [comment]
With proper parameters, we can get the mIoU 33%+ with CSW.

**Comment by Ze-Yang** [comment]
I just follow the hyperparameters in your script. Do you mean those hyperparameters are not the ones you use to report the results in the paper?

**Comment by HieuPhan33** [comment]
This is the log file that I achieve for 100-10. 
[2022-03-26_ade_100-10_REMINDER.csv](https://github.com/HieuPhan33/REMINDER/files/9399683/2022-03-26_ade_100-10_REMINDER.csv)

I will double-check the hyper-parameters and re-run 100-10.
Then I'll share with you the results.

**Comment by Ze-Yang** [comment]
Based on my experiments, the CSW loss does not contribute to 100-10 setting. The improvement on new classes against PLOP is **actually from the training strategies**, i.e., `batch size=10`, `max_epoch=70`, which in fact sacrifices the old class performance by 1 percent point. I also test these training strategies (`batch size=10`, `max_epoch=70`) on PLOP's codebase. The resutls `PLOP + bs 10 + 70 epochs` are similar to `REMINDER w/o csw`. In addition, I roughly compare the code differences between REMINDER and PLOP. I can only notice the differences in the CSW loss and training hyperparameters.
|| 0-100 | 101-150 | all |
|--|:--:|:--:|:--:|
| PLOP (reproduced) | 39.29 | 14.48 | 31.07 |
| PLOP + bs 10 + 70 epochs | 38.35 | 20.64 | 32.48 |
| REMINDER w/o csw | 38.64 | 20.58 | 32.66 |
| REMINDER w/ csw | 38.71 | 20.33 | 32.63 |

**Comment by Ze-Yang** [comment]
Hi @HieuPhan33, here are the results for 100-50.
|| 0-100 | 101-150 | all |
|--|:--:|:--:|:--:|
| REMINDER w/o csw | 41.83 | 22.47 | 35.42 |
| REMINDER w/ csw | 41.85 | 22.29 | 35.37 |
